pkg/security/ebpf/c/include/hooks/setsockopt.h

```diff
@@ -3,64 +3,152 @@
 
 #include "constants/syscall_macro.h"
 #include "helpers/syscalls.h"
-
-long __attribute__((always_inline)) trace__sys_setsock_opt(u8 async, int socket, int level, int optname) {
+#include <uapi/linux/filter.h>
+long __attribute__((always_inline)) trace__sys_setsock_opt(u8 async, int socket_fd, int level, int optname) {
     if (is_discarded_by_pid()) {
         return 0;
     }
-
-    struct policy_t policy = fetch_policy(EVENT_SETSOCKOPT);
-    struct syscall_cache_t syscall = {
-        .type = EVENT_SETSOCKOPT,
-        .policy = policy,
-        .async = async,
-        .setsockopt = {
-            .level = level,
-            .optname = optname,
+    switch (optname) {
+        case SO_ATTACH_FILTER: {
+            struct policy_t policy = fetch_policy(EVENT_SETSOCKOPT);
+            struct syscall_cache_t syscall = {
+                .type = EVENT_SETSOCKOPT,
+                .policy = policy,
+                .async = async,
+                .setsockopt = {
+                    .level = level,
+                    .optname = optname,
+                }
+            };
+
+            cache_syscall(&syscall);
+            return 0;
         }
-    };
-
-    cache_syscall(&syscall);
-
-    return 0;
+        default:
+            return 0; // unsupported optname
+    }
 }
 
 int __attribute__((always_inline)) sys_set_sock_opt_ret(void *ctx, int retval) {
     struct syscall_cache_t *syscall = pop_syscall(EVENT_SETSOCKOPT);
-
     if (!syscall) {
         return 0;
     }
+    int key = 0;
+    struct setsockopt_event_t *event = bpf_map_lookup_elem(&setsockopt_event,&key);
```

Comment:
Is it possible to use `__builtin_memset` on the setsockopt event here in order to reset the event fields and make sure we're not using any data that was previously set for another event?